velox/connectors/hive/storage_adapters/abfs/AzureClientProviderFactories.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/hive/storage_adapters/abfs/AzureClientProviderFactories.h"
+#include "velox/connectors/hive/storage_adapters/abfs/AzureClientProviders.h"
+
+#include <folly/Synchronized.h>
+
+namespace facebook::velox::filesystems {
+
+namespace {
+
+folly::Synchronized<
+    std::unordered_map<std::string, AzureClientProviderFactory>>&
+azureClientFactoryRegistry() {
+  static folly::Synchronized<
+      std::unordered_map<std::string, AzureClientProviderFactory>>
+      factories;
+  return factories;
+}
+
+} // namespace
+
+void AzureClientProviderFactories::registerFactory(
+    const std::string& account,
+    const AzureClientProviderFactory& factory) {
+  azureClientFactoryRegistry().withWLock(
```

Comment:
We should log if a factory is being overridden for a specific account. It will help aid debugging.